packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -95,6 +95,33 @@ export function useFocusFirstElement( clientId ) {
 			return;
 		}
 
+		// Check to see if Block contains focussable element before a generic caret insert.
+		if ( ! ref.current.getAttribute( 'contenteditable' ) ) {
+			const whitelistTags = [ 'a', 'button', 'td' ];
+			let focusElements;
+			let firstFocusElements = true;
+			// In the event a Block has children, focus will find the next element with tabindex which would happen to be the wrapping element of a child Block. This ensures we only focus on whitelisted tag such as link/button.
+			do {
+				// The first element may be okay to focus. Go ahead and check.
```

Comment:
In the event of the Table Block and maybe others, the first td could be a good point to focus. I changed the code slightly to always check the first element.